### PR TITLE
2208 collapsed footer link groups not focusable

### PR DESCRIPTION
**Pull request by gd2910**
## Summary of the changes
Gave the parent ic-section-container of grouped-links a tabindex="0" to allow tabbing.

Noticed a couple of failing a11y tests when writing a cypress test for the change
 - link-group had a tabindex of 1
 - grouped-links in mobile view needed child aria-role of 'listitem'

## Related issue
#2208

### CODE REVIEW

File: packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx [hunk 1]
```diff
@@ -70,13 +73,32 @@ describe("IcFooter end-to-end, visual regression and a11y tests", () => {
     });
   });
 
+  it.only("should tab grouped links and open with keyboard in mobile viewport", () => {
```
Comment (evenstensberg) [inline packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx:76]: Only test needs to be removed

**Review by evenstensberg — COMMENTED** [review]
Probably too early for a review but the test.only needs to be removed and the latest commit message needs rewording (git rebase -I HEAD~1)

**Comment by gd2910** [comment]
Hi @evenstensberg @GCHQ-Developer-530 , still working on this PR, needed to bring out of draft to run CI actions on the branch. I'll notify when ready for review.

**Comment by gd2910** [comment]
Ready for review.

**Comment by evenstensberg** [comment]
I can’t approve due security but lgtm

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by GCHQ-Developer-741 — APPROVED** [review]

**Pull request merged**